core/domain/fs_domain.py

```diff
@@ -107,26 +117,70 @@ def _validate_entity_parameters(self, entity_name, entity_id):
             raise utils.ValidationError('Entity id cannot be empty')
 
     @property
-    def assets_path(self):
+    def assets_path(self) -> str:
         """Returns the path of the parent folder of assets.
 
         Returns:
             str. The path.
         """
+
         return self._assets_path
 
+    def _get_gcs_file_url(self, filepath: str) -> str:
+        """Returns the constructed GCS file URL.
+
+        Args:
+            filepath: str. The path to the relevant file within the entity's
+                assets folder.
+
+        Returns:
+            str. The GCS file URL.
+        """
+
+        # Upload to GCS bucket with filepath
+        # "<entity>/<entity-id>/assets/<filepath>".
+        raise NotImplementedError
+
+    def isfile(self, filepath: str) -> bool:
+        """Checks if the file with the given filepath exists in the GCS.
+
+        Args:
+            filepath: str. The path to the relevant file within the entity's
+                assets folder.
+
+        Returns:
+            bool. Whether the file exists in GCS.
+        """
+
+        raise NotImplementedError
+
+    def get(self, filepath: str) -> FileStream | None: # pylint: disable=missing-docstring
```

Comment:
add docstring ditto elsewhere

Reply:
I am getting indentation errors while adding docstrings, thats why I ignored them.

Reply:
Can you fix the errors somehow?